Doing things the right way, as per ActiveRecord

weshatheleopard opened the pull request.

Commit:
Copying over a few methods that are defined on AR::AssociationReflection.
Absence of these on EnumerationReflection causes ransack_ui to barf.

Commit:
I blame Mark, he told me it shold be false! :-b

hubrix committed:
comment out signing key

it seems broken

Errno::ENOENT: No such file or directory - gem-private_key.pem
An error occurred while installing power_enum (1.3.0), and Bundler cannot continue.
Make sure that `gem install power_enum -v '1.3.0'` succeeds before bundling.

albertosaurus commented on power_enum.gemspec:
```diff
@@ -38,7 +38,7 @@ Gem::Specification.new do |s|
   s.require_paths = ["lib"]
   s.rubygems_version = "1.8.25"
-  s.signing_key = "gem-private_key.pem"
+  #s.signing_key = "gem-private_key.pem"
```
Please don't do that.
1. This gem is signed with my private key when I build it.  If you're going to go through the trouble of trying to install it locally, generate a set of signing keys.  See this: http://docs.rubygems.org/read/chapter/21
2. For better or worse, I use jeweler to develop the gem.  The gemspec is generated by running `rake gemspec`.  Any changes made to the gemspec get overwritten.

albertosaurus replied:
I have updated the README to better explain how setup PowerEnum in high security mode.

albertosaurus commented on lib/power_enum/reflection.rb:
```diff
@@ -95,11 +95,11 @@ def chain
+      macro == :belongs_to
     end
 
+    alias :source_macro :macro
```
This line of code breaks four unit tests.  The `source_macro` for PowerEnum is defined as `:belongs_to`.  As mentioned in the RDoc, it's necessary to make joins work.

albertosaurus replied:
Note: https://travis-ci.org/albertosaurus/power_enum/builds/6266183

albertosaurus commented on lib/power_enum/reflection.rb:
```diff
@@ -95,11 +95,11 @@ def chain
+    # Returns +true+ if +self+ is a +belongs_to+ reflection.
     def belongs_to?
-      false
+      macro == :belongs_to
```
Not going to work.

albertosaurus replied:
I think you're better off explicitly defining `macro`.  Something like

``` ruby
def macro
  :belongs_to
end
```

Having said that, I don't know off the top of my head what the second-order implications of that are going to be.

Commit:
Add PowerEnum::Reflection::EnumerationReflection::polymorphic?() function
that just returns false for power enum reflections.

This is required (at least in Rails4) to do queries like
  where( enum: Enum[value] )
